packages/forms/src/styled/file-list/StyledFile.ts

```diff
@@ -5,30 +5,104 @@
+import { rgba } from 'polished';
 import { retrieveComponentStyles, DEFAULT_THEME, getColor } from '@zendeskgarden/react-theming';
+import { StyledFileClose } from './StyledFileClose';
 
 const COMPONENT_ID = 'forms.file';
 
+const colorStyles = (props: IStyledFileProps & ThemeProps<DefaultTheme>) => {
+  const borderColor = getColor('neutralHue', 300, props.theme);
+  const focusBorderColor = getColor('primaryHue', 600, props.theme);
+  const foregroundColor = getColor('neutralHue', 800, props.theme);
+  const boxShadow = `
+    ${props.focusInset ? 'inset' : ''}
+    ${props.theme.shadows.md(rgba(focusBorderColor!, 0.35))}`;
+
+  return `
+    border-color: ${borderColor};
+    color: ${foregroundColor};
+
+    &:focus {
+      box-shadow: ${boxShadow};
+      border-color: ${focusBorderColor};
+    }
+  `;
+};
+
+const sizeStyles = (props: IStyledFileProps & ThemeProps<DefaultTheme>) => {
+  const size = `${props.theme.space.base * (props.isCompact ? 7 : 10)}px`;
+  const spacing = `${props.theme.space.base * 3}px`;
+
+  return `
+    box-sizing: border-box;
+    border: ${props.theme.borders.sm};
+    border-radius: ${props.theme.borderRadii.md};
+    padding: 0 ${spacing};
+    height: ${size};
+    font-size: ${props.theme.fontSizes.md};
+
+    & > span {
+      width: 100%;
+    }
+
+    & > ${StyledFileClose} {
+      width: ${size};
+      height: ${size};
+      margin-${props.theme.rtl ? 'left' : 'right'}: -${spacing};
+    }
+  `;
+};
+
 interface IStyledFileProps {
-  isDragging?: boolean;
   isCompact?: boolean;
+  focusInset?: boolean;
 }
 
 export const StyledFile = styled.div.attrs({
   'data-garden-id': COMPONENT_ID,
   'data-garden-version': PACKAGE_VERSION
 })<IStyledFileProps>`
-  display: inline-flex;
+  display: flex;
   position: relative;
   flex-wrap: nowrap;
   align-items: center;
-  border: ${props => `${props.theme.borders.sm} ${getColor('neutralHue', 300, props.theme)}`};
-  border-radius: ${props => props.theme.borderRadii.md};
-  padding: 0 ${props => props.theme.space.base * 3}px;
-  height: ${props => (props.isCompact ? 30 : 38)}px;
-  color: ${props => getColor('neutralHue', 800, props.theme)};
-  font-size: ${props => props.theme.fontSizes.md};
+
+  ${sizeStyles};
+
+  &:focus {
+    outline: none;
+  }
+
+  ${colorStyles};
+
+  & > span {
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+  }
+
+  & > [role='progressbar'] {
+    position: absolute;
+    bottom: 0;
+    left: 0;
+    transition: opacity 0.2s ease-in-out;
+    margin: 0;
+    border-top-left-radius: 0;
+    border-top-right-radius: 0;
+    width: 100%;
+
+    & > div {
+      /* stylelint-disable-next-line property-no-unknown, property-case */
+      border-top-${props => (props.theme.rtl ? 'right' : 'left')}-radius: 0;
+    }
+  }
+
+  & > [role='progressbar'][aria-valuenow='0'],
+  & > [role='progressbar'][aria-valuenow='100'] {
+    opacity: 0;
+  }
```

Comment:
Per component RFC, styling tweaks for the contained `Progress` component are meant to be delivered as part of the `File` component itself.